server/src/main/java/org/opensearch/rest/NamedRoute.java

```
@@ -9,47 +9,168 @@
+        public Builder method(RestRequest.Method method) {
+            this.method = method;
+            return this;
+        }
+
+        /**
+         * Sets the URL path for the route.
+         *
+         * @param path the URL path for the route
+         * @return the builder instance
+         */
+        public Builder path(String path) {
+            this.path = path;
+            return this;
+        }
+
+        /**
+         * Sets the name for the route.
+         *
+         * @param name the name for the route
+         * @return the builder instance
+         */
+        public Builder uniqueName(String name) {
+            this.uniqueName = name;
+            return this;
+        }
+
+        /**
+         * Sets the legacy action names for the route.
+         *
+         * @param legacyActionNames the legacy action names for the route
+         * @return the builder instance
+         */
+        public Builder legacyActionNames(Set<String> legacyActionNames) {
+            this.legacyActionNames.addAll(validateLegacyActionNames(legacyActionNames));
+            return this;
+        }
+
+        /**
+         * Sets the handler for this route
+         *
+         * @param handler the handler for this route
+         * @return the builder instance
+         */
+        public Builder handler(Function<RestRequest, RestResponse> handler) {
+            if (handler == null) {
+                throw new OpenSearchException("Route handler must not be null.");
+            }
+            this.handler = handler;
+            return this;
         }
-        return routeName.matches(VALID_ACTION_NAME_PATTERN);
+
+        /**
+         * Builds a new instance of {@link NamedRoute} based on the provided parameters.
+         *
+         * @return a new instance of {@link NamedRoute}
+         * @throws OpenSearchException if the route name is invalid
+         */
+        public NamedRoute build() {
+            return new NamedRoute(this);
```

You should validate that all the required values have been set and throw an exception if they have not.  For example, `new NamedRoute.Builder().build()` should throw an exception noting that a method is required, etc.